src/utils/debounce.js

```diff
@@ -0,0 +1,40 @@
+const now = Date.now || function _now() {
+  return new Date().getTime();
+};
+
+export default function debounce(fn, wait, immediate) {
+  let timeout;
+  let args;
+  let context;
+  let timestamp;
+  let result;
+
+  const later = () => {
+    const last = now() - timestamp;
+
+    if (last < wait && last >= 0) {
```

Comment:
This must be covering an interesting edge case. Maybe there are scenarios where `later()` gets called before `wait` has elapsed?